native/src/main/java/io/ballerina/stdlib/grpc/builder/syntaxtree/utils/MessageUtils.java

```diff
@@ -156,6 +159,9 @@ private static Type getMessageType(Message message) {
                         case "Duration":
                             messageRecord.addOptionalCustomField("time:Seconds", field.getFieldName());
                             break;
+                        case "Struct":
+                            messageRecord.addOptionalCustomField("map<anydata>", field.getFieldName());
```

Comment:
Let's create a new method called `addOptionalMapField` similar to `addOptionalArrayField`[[1]](https://github.com/ballerina-platform/module-ballerina-grpc/blob/d269ad779461978fe548d2817a8103cc8b181a12/native/src/main/java/io/ballerina/stdlib/grpc/builder/syntaxtree/components/Record.java#L137).

`addOptionalCustomField` function is intended for types such as time:Utc.

[1]https://github.com/ballerina-platform/module-ballerina-grpc/blob/d269ad779461978fe548d2817a8103cc8b181a12/native/src/main/java/io/ballerina/stdlib/grpc/builder/syntaxtree/components/Record.java#L137

Reply:
Changed in https://github.com/ballerina-platform/module-ballerina-grpc/pull/368/commits/67eb9b0783f2bd76dfa9565bd36c53e66aec9d88